dashboard/src/components/AppList/AppList.v2.test.tsx

```diff
@@ -0,0 +1,260 @@
+import { mount, shallow } from "enzyme";
+import context from "jest-plugin-context";
+import * as React from "react";
+
+import SearchFilter from "components/SearchFilter/SearchFilter.v2";
+import { BrowserRouter as Router } from "react-router-dom";
+import itBehavesLike from "../../shared/specs";
+import { IAppOverview, IAppState } from "../../shared/types";
+import Alert from "../js/Alert";
+import AppList, { IAppListProps } from "./AppList.v2";
+import AppListItem from "./AppListItem.v2";
+import CustomResourceListItem from "./CustomResourceListItem.v2";
+
+let props = {} as any;
+
+const defaultProps: IAppListProps = {
+  apps: {} as IAppState,
+  filter: "",
+  namespace: "default",
+  cluster: "defaultc",
+  pushSearchFilter: jest.fn(),
+  fetchAppsWithUpdateInfo: jest.fn(),
+  getCustomResources: jest.fn(),
+  isFetchingResources: false,
+  customResources: [],
+  csvs: [],
+  featureFlags: { operators: true, additionalClusters: [], ui: "hex" },
+  appVersion: "v1.0.0",
+};
+
+context("when changing props", () => {
+  it("should fetch apps in the new namespace", async () => {
+    const fetchAppsWithUpdateInfo = jest.fn();
+    const getCustomResources = jest.fn();
+    mount(
+      <Router>
+        <AppList
+          {...defaultProps}
+          fetchAppsWithUpdateInfo={fetchAppsWithUpdateInfo}
+          getCustomResources={getCustomResources}
+          namespace="foo"
+        />
+      </Router>,
+    );
+    expect(fetchAppsWithUpdateInfo).toHaveBeenCalledWith("foo", true);
+    expect(getCustomResources).toHaveBeenCalledWith("foo");
+  });
+
+  it("should update the filter", () => {
+    const wrapper = mount(
+      <Router>
+        <AppList {...defaultProps} filter="foo" />
+      </Router>,
+    );
+    expect(wrapper.find(SearchFilter).prop("value")).toEqual("foo");
+  });
+});
+
+context("while fetching apps", () => {
+  props = { ...defaultProps, apps: { isFetching: true } };
+
+  itBehavesLike("aLoadingComponent", { component: AppList, props });
+  itBehavesLike("aLoadingComponent", {
+    component: AppList,
+    props: { ...defaultProps, isFetchingResources: true },
+  });
+
+  it("matches the snapshot", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper).toMatchSnapshot();
```

Comment:
I usually use `toMatchSnapshot` only with small portions of the components to avoid failing tests due to minimal changes. Just a comment on this :)

Reply:
yep, this test was copied from the one existing but it's pretty useless.